cmake/BuildLuaCAPITests.cmake

```diff
@@ -0,0 +1,56 @@
+set(LUA_C_API_TESTS_TARGET lua-c-api-tests)
+# Beware: OSS Fuzz build script searches tests in a directory
+# `${PROJECT_BINARY_DIR}/test/fuzz`.
+set(LUA_C_API_TESTS_PREFIX ${PROJECT_BINARY_DIR}/test/fuzz/${LUA_C_API_TESTS_TARGET})
+
+string(JOIN ";" LUA_C_API_TESTS
+  ffi_cdef_proto_test
+  lua_dump_test
+  luaL_gsub_test
+  luaL_loadbuffer_proto_test
+  luaL_loadbufferx_test
+  lua_load_test
+  luaL_traceback_test
+  torture_test
+)
+
+include(ExternalProject)
+
+ExternalProject_Add(${LUA_C_API_TESTS_TARGET}
+  PREFIX ${LUA_C_API_TESTS_TARGET}
+  GIT_REPOSITORY https://github.com/ligurio/lua-c-api-tests.git
+  # FIXME: Depends on to be merged changes,
+  # https://github.com/ligurio/lua-c-api-tests/pull/113.
+  GIT_TAG ligurio/gh-xxxx-set-liblua-outside
```

Comment:
This version leads to the following error:

```
CMake Error at tests/capi/CMakeLists.txt:69 (add_executable):
  Target "luaL_loadbuffer_proto_test" links to item " -g" which has leading
  or trailing whitespace.  This is now an error according to policy CMP0004.
Call Stack (most recent call first):
  tests/capi/luaL_loadbuffer_proto/CMakeLists.txt:29 (create_test)
```

The reason is that we copy the `CMAKE_C_FLAGS_DEBUG` value to `LDFLAGS` [here](https://github.com/ligurio/lua-c-api-tests/blob/ef74f0558bb913edef85e7a774fada4ecc82247b/tests/capi/CMakeLists.txt#L30), which [contains](https://github.com/tarantool/luajit/blob/3f4f46a08e57f96377955e057b1b59bede15c473/CMakeLists.txt#L112) ` -g`. Looks like `CMAKE_C_FLAGS_DEBUG` shouldn't be copied to the `LDFLAGS` at all.

Reply:
Fixed in PR https://github.com/ligurio/lua-c-api-tests/pull/113, commit "tests: fix leading whitespace in linker flags".